### ISSUE TITLE
BatArray.sum/fsum don't behave like the recently fixed BatList.sum/fsum

### ISSUE THREAD

**Issue by UnixJunkie** [ocaml-batteries-team/batteries-included#919]
```
utop # BatArray.sum [||];;
Exception: Invalid_argument "Array.reduce: empty array".
utop # BatArray.fsum [||];;
Exception: Invalid_argument "Array.reduce: empty array".

```

**Comment by rixed** [comment]
Should be though: https://github.com/ocaml-batteries-team/batteries-included/commit/3c323ef1707b9f89c2d9c8694f0163853b21e61c#diff-adc58a349b1a0a37e681dc37960cc8df

what's going on?

**Comment by UnixJunkie** [comment]
Ok, my mistake. It was indeed included in your PR.
